crates/sncast/tests/data/scripts/deploy/src/invalid_class_hash.cairo

```diff
@@ -0,0 +1,25 @@
+use sncast_std::{get_nonce, deploy, DeployResult, ScriptCommandError, ProviderError, StarknetError};
+
+use starknet::{ClassHash, Felt252TryIntoClassHash};
+use traits::Into;
+
+fn main() {
+    let max_fee = 99999999999999999;
+    let salt = 0x3;
+
+    let class_hash: ClassHash = 0xdddd.try_into().expect('Invalid class hash value');
+
+    let deploy_nonce = get_nonce('pending');
+    let deploy_result = deploy(
+        class_hash,
+        array![0x2, 0x2, 0x0],
+        Option::Some(salt),
+        true,
+        Option::Some(max_fee),
+        Option::Some(deploy_nonce)
+    )
+        .unwrap_err();
+
```

Comment:
should we assert the error here as well?

Reply:
probably applicable to other tests as well

Reply:
In some tests, I assert the error like here:
```
assert(
    ScriptCommandError::ProviderError(
        ProviderError::StarknetError(StarknetError::InvalidTransactionNonce)
    ) == deploy_result,
    'ohno'
)
```

But it becomes more complex when the enum variant contains a struct like here:
```
pub enum StarknetError {
  ContractError: ErrorData,
}

pub struct ErrorData {
    msg: ByteArray
}
...
```

Reply:
but I guess it's enough to assert the output produced with `println!("{:?}", result)`